### ISSUE TITLE
Error when resolving `.length()` expression in 2.5.0

### ISSUE THREAD

**Issue by lukas-krecan** [json-path/JsonPath#650]
After upgrading to 2.5.0, this code started to return strange results

```java
        String json = "{\n" +
            "    \"store\": {\n" +
            "        \"book\": [\n" +
            "            {\n" +
            "                \"category\": \"reference\"\n" +
            "            },\n" +
            "            {\n" +
            "                \"category\": \"fiction\"\n" +
            "            },\n" +
            "            {\n" +
            "                \"category\": \"fiction\"\n" +
            "            },\n" +
            "            {\n" +
            "                \"category\": \"fiction\"\n" +
            "            }\n" +
            "        ]\n" +
            "    },\n" +
            "    \"expensive\": 10\n" +
            "}";


        Object result = using(Configuration.defaultConfiguration())
            .parse(json)
            .read("$..book.length()");
        System.out.println(result);
```

In 2.4.0 it returns an array containing number 4, in 2.5.0 it returns 2.

**Comment by kallestenflo** [comment]
This regression was intoduced by #377. @mgreenwood1001, interested in taking a look?

**Comment by mgreenwood1001** [comment]
I'll take a look

**Comment by mgreenwood1001** [comment]
@kallestenflo please see the bugfix for length()

**Comment by rahulgandhi** [comment]
@kallestenflo @mgreenwood1001 Thanks for the fix.
Given that https://github.com/json-path/JsonPath/pull/651 has been merged, when can we expect the fix to be available in a release version?

**Comment by kallestenflo** [comment]
Fixed in 2.6.0
